rollout/service.go

```diff
@@ -284,6 +285,27 @@ func (c *rolloutContext) reconcileStableAndCanaryService() error {
 	return nil
 }
 
+func (c *rolloutContext) getCanaryReplicaSet() (*appsv1.ReplicaSet, error) {
+	if c.rollout.Spec.Strategy.Canary == nil {
+		return nil, nil
+	}
+	svc, err := c.servicesLister.Services(c.rollout.Namespace).Get(c.rollout.Spec.Strategy.Canary.CanaryService)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
If there is no canaryService, the rollout will be stuck too?

Reply:
yes, but that's a negative case.

but this is really a goot catch! i should check whether the canary service is an empty string first.